plugins/inputs/modbus/modbus.go

```diff
@@ -345,6 +386,9 @@ func (m *Modbus) gatherRequestsCoil(requests []request) error {
 			request.fields[i].value = uint16((bytes[idx] >> bit) & 0x01)
 			m.Log.Debugf("  field %s with bit %d @ byte %d: %v --> %v", field.name, bit, idx, (bytes[idx]>>bit)&0x01, request.fields[i].value)
 		}
+
+		// Some (serial) devices require a pause between requests...
+		time.Sleep(time.Until(nextRequest))
```

Comment:
There's an opportunity here for the user to configure more sleep time than the total collection interval. Should we warn them if for example they have 10 requests with 1 second sleep between and that won't fit into a 5 second collection interval?

(same goes for discrete, holding, input, etc. in addition to coil)

Reply:
The correct solution would be to allow the plugin to read the configured interval of the plugin. This would also ease the life of plugins that need to specify the collection interval for reading.

The problem with your suggestion is that there might be long intervals (e.g. 15 minutes) with long requests adding up to e.g. 1 minute rendering this a valid use-case. As long as we cannot know the collection interval we cannot decide on the validity of the sleep interval, potentially either classifying valid use-cases as errors or vice versa.

Reply:
It would be nice for the plugin to have some way to learn what its collection interval is, but since that doesn't exist now, let's also document this so the user is aware of it. Something like, "make sure not to add pauses that are too long or the plugin will miss the next collection interval"

Reply:
That's also a thing I believe is missing from the snmp input docs (or any other that has timeouts)

Reply:
@reimda I agree and will add some words there.